src/tests/Regressions/coreclr/GitHub_43763/test43763.cs

```diff
@@ -0,0 +1,48 @@
+﻿class Program
+{
+    static int Main(string[] args)
+    {
+        System.Console.WriteLine(System.Runtime.InteropServices.RuntimeInformation.FrameworkDescription);
+        CallC();
+        CallB();
+        CallC2();
+        CallB2();
+
+        return 100;
+    }
+
+    static void CallB() => new B();
+    static void CallC() => new C();
+    static void CallB2() => new B2();
+    static void CallC2() => new C2();
+}
+
+abstract class A<T>
+{
+    public abstract A<T> M();
+}
+
+abstract class A2<T>
+{
+    public abstract A2<T> M<U>();
+}
+
+class B : A<string>
+{
+    public override B M() => new B();
+}
+
+class B2 : A2<string>
+{
+    public override B2 M<U>() => new B2();
+}
+
+class C : A<int>
+{
+    public override C M() => new C();
+}
+
+class C2 : A2<int>
```

Comment:
@janvorli, just curious, would the fix also handle `class D : C2` for this `C2` deriving generic `A2<>`?

Reply:
@am11 the code only checks the return type compatibility between the current and parent class method. So for the case you've mentioned, it would check for compatibility between D.M and C2.M return types where there is no canon type and thus there was no problem.